packages/debugger/lib/data/sagas/index.js

```diff
@@ -1298,6 +1042,323 @@ function fetchBasePath(
+    debug("request received");
+    let request = result.value;
+    let response;
+    switch (request.type) {
+      //skip storage case, it won't happen here
+      case "code":
+        response = yield* requestCode(request.address);
+        break;
+      default:
+        debug("unrecognized request type!");
+    }
+    debug("sending response");
+    result = decoder.next(response);
+  }
+  //at this point, result.value holds the final value
+  debug("done decoding");
+  debug("decoded value: %O", result.value);
+  return result.value;
+}
+
+//NOTE: calling this *can* add a new instance, which will not
+//go away on a reset!  Yes, this is a little weird, but we
+//decided this is OK for now
+function* requestCode(address) {
+  const NO_CODE = new Uint8Array(); //empty array
+  const blockNumber = yield select(data.views.blockNumber);
+  const instances = yield select(data.views.instances);
+
+  if (address in instances) {
+    return instances[address];
+  } else if (address === Codec.Evm.Utils.ZERO_ADDRESS) {
+    //HACK: to avoid displaying the zero address to the user as an
+    //affected address just because they decoded a contract or external
+    //function variable that hadn't been initialized yet, we give the
+    //zero address's codelessness its own private cache :P
+    return NO_CODE;
+  } else {
+    //I don't want to write a new web3 saga, so let's just use
+    //obtainBinaries with a one-element array
+    debug("fetching binary");
+    let binary = (yield* web3.obtainBinaries([address], blockNumber))[0];
+    debug("adding instance");
+    yield* evm.addInstance(address, binary);
+    return Codec.Conversion.toBytes(binary);
+  }
+}
+
+export function* reset() {
+  yield put(actions.reset());
+}
+
+export function* recordAllocations() {
+  const contracts = yield select(data.views.contractAllocationInfo);
+  const referenceDeclarations = yield select(data.views.referenceDeclarations);
+  const userDefinedTypesByCompilation = yield select(
+    data.views.userDefinedTypesByCompilation
+  );
+  const userDefinedTypes = yield select(data.views.userDefinedTypes);
+  const storageAllocations = Codec.Storage.Allocate.getStorageAllocations(
+    userDefinedTypesByCompilation
+  );
+  const memoryAllocations =
+    Codec.Memory.Allocate.getMemoryAllocations(userDefinedTypes);
+  const abiAllocations =
+    Codec.AbiData.Allocate.getAbiAllocations(userDefinedTypes);
+  const calldataAllocations = Codec.AbiData.Allocate.getCalldataAllocations(
+    contracts,
+    referenceDeclarations,
+    userDefinedTypes,
+    abiAllocations
+  );
+  const returndataAllocations = Codec.AbiData.Allocate.getReturndataAllocations(
+    contracts,
+    referenceDeclarations,
+    userDefinedTypes,
+    abiAllocations
+  );
+  const eventAllocations = Codec.AbiData.Allocate.getEventAllocations(
+    contracts,
+    referenceDeclarations,
+    userDefinedTypes,
+    abiAllocations,
+    true //signals that we are allowing events from constructors
```

Comment:
I really feel like these kinds of positional arguments are only increasing the fragility of this codebase, even with the comment. Won't block merge over it, but I will encourage @cds-amal to do so instead ;)

Reply:
Maybe we just need to spend some time in the future doing stylistic cleanup.

Reply:
I don't think those changes should hold back this feature, but, the positional style should be addressed in follow-up work.